File: Examples2018/BcastRecSinBcastStatReg/app/src/main/java/course/examples/broadcastreceiver/singlebroadcaststaticregistration/SimpleBroadcastActivity.java
package course.examples.broadcastreceiver.singlebroadcaststaticregistration;

import android.Manifest;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class SimpleBroadcastActivity extends Activity {

    private static final String CUSTOM_INTENT = "course.examples.broadcastreceiver.singlebroadcaststaticregistration.SHOW_TOAST";
    private static final String TAG = "SimpleBroadcastActivity";

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);
    }

    public void onClick(@SuppressWarnings("unused") View v) {
        Log.i(TAG, "Broadcast sent");

        Intent intent = new Intent(CUSTOM_INTENT);
        intent.setPackage("course.examples.broadcastreceiver.singlebroadcaststaticregistration");
        sendBroadcast(intent, Manifest.permission.VIBRATE);
    }
}

File: ExamplesKotlin/BcastRecSinBcastStatReg/app/src/main/java/course/examples/broadcastreceiver/singlebroadcaststaticregistration/SimpleBroadcastActivity.kt
package course.examples.broadcastreceiver.singlebroadcaststaticregistration

import android.Manifest
import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View

class SimpleBroadcastActivity : Activity() {


    companion object {

        private const val CUSTOM_INTENT =
            "course.examples.broadcastreceiver.singlebroadcaststaticregistration.SHOW_TOAST"
        private const val TAG = "SimpleBroadcastActivity"
    }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)
    }

    @SuppressWarnings("UnusedParameters")
    fun onClick(view: View) {
        Log.i(TAG, "Broadcast sent")

        val intent = Intent(CUSTOM_INTENT)
        intent.setPackage("course.examples.broadcastreceiver.singlebroadcaststaticregistration")
        sendBroadcast(intent, Manifest.permission.VIBRATE)
    }
}